app/models/mixins/retirement_mixin.rb

```diff
@@ -39,6 +39,7 @@ def retires_on=(timestamp)
     if retires_on != timestamp
       self.retired              = false if timestamp.nil? || (timestamp.to_date > Date.today)
       self.retirement_last_warn = nil # Reset so that a new warning can be sent out when the time is right
+      self.retirement_state     = 'retired' if ( not timestamp.nil? && timestamp.to_date <= Date.today) 
```

Comment:
Style preference (also listed by rubocop): Prefer `!` over `not`.  Also remove the parens.

Or instead, it might cleaner to just do `if timestamp && timestamp.to_date <= Date.today`